lib/request.js

```diff
@@ -395,9 +397,8 @@ module.exports = {
 
   get ips() {
     var proxy = this.app.proxy;
-    var val = this.get('X-Forwarded-For');
-    return proxy && val
-      ? val.split(/\s*,\s*/)
+    return proxy
+      ? forwarded(this.req)
```

Comment:
Depending on what you want `ips` to be, ideally they should get cut off at the trust endpoint. You can remove a dependency with using https://github.com/jshttp/proxy-addr#proxyaddrallreq-trust
```js
return proxyAddr.all(this,req, this.app._proxyTrust);
```